udata/api/__init__.py

```diff
@@ -120,13 +120,19 @@ def search_parser(self, adapter, paginate=True):
         # q parameter
         parser.add_argument('q', type=str, location='args',
                             help='The search query')
+        facets = adapter.facets.keys()
+        if facets:
+            parser.add_argument('facets', type=str, location='args',
+                                choices=['all'] + facets, action='append',
+                                help='Selected facets to fetch')
         # Add facets filters arguments
         for name, facet in adapter.facets.items():
```

Comment:
no, there is 2 aspecst on facets: the filtering was already implemented. This PR only expose the facet querying and results.

These lines were not touched because they are doing the filtering part.